src/Localization/Localization/src/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -228,17 +233,21 @@ protected virtual ResourceLocationAttribute GetResourceLocationAttribute(Assembl
         /// <param name="assembly">The assembly to get a <see cref="RootNamespaceAttribute"/> from.</param>
         /// <returns>The <see cref="RootNamespaceAttribute"/> associated with the given <see cref="Assembly"/>.</returns>
         /// <remarks>This method is protected and virtual for testing purposes only.</remarks>
-        protected virtual RootNamespaceAttribute GetRootNamespaceAttribute(Assembly assembly)
+        protected virtual RootNamespaceAttribute? GetRootNamespaceAttribute(Assembly assembly)
         {
             return assembly.GetCustomAttribute<RootNamespaceAttribute>();
         }
 
-        private string GetRootNamespace(Assembly assembly)
+        private string? GetRootNamespace(Assembly assembly)
         {
             var rootNamespaceAttribute = GetRootNamespaceAttribute(assembly);
 
-            return rootNamespaceAttribute?.RootNamespace ??
-                new AssemblyName(assembly.FullName).Name;
+            if (rootNamespaceAttribute != null)
+            {
+                return rootNamespaceAttribute.RootNamespace;
+            }
+
+            return assembly.GetName().Name;
```

Comment:
Just curious: Did the compiler complain about the previoius expression? If yes, what was it's concern?

Reply:
Both `FullName` and `Name` are nullable and `AssemblyName`'s ctor  does not like nulls. This had fewer null checks